src/storage.ts

```diff
@@ -1,13 +1,59 @@
-type IStorage = {
-    count: number;
-};
-
-const defaultStorage: IStorage = {
-    count: 0,
-};
-
-export const storage = {
-    get: (): Promise<IStorage> =>
-        chrome.storage.sync.get(defaultStorage) as Promise<IStorage>,
-    set: (value: IStorage): Promise<void> => chrome.storage.sync.set(value),
-};
+import { writable, type Writable } from 'svelte/store';
+
+/**
+ * Creates a persistent Svelte store backed by Chrome's sync storage.
+ * @template T The type of the store's value
+ * @param key The key to use in Chrome's storage
+ * @param initialValue The initial value of the store
+ * @returns A writable Svelte store
+ */
+export function persistentStore<T>(key: string, initialValue: T): Writable<T> {
+    const store = writable(initialValue);
+    let storeValues: T[] = [];
+    let chromeValues: T[] = [];
+
+    function watchStore() {
+        store.subscribe((value) => {
+            // Prevent circular updates
+            if (chromeValues && value === chromeValues[0]) {
```

Comment:
```suggestion
            if (value === chromeValues[0]) {
```

Reply:
I think this could trip an edge case where we actually store `undefined`

Reply:
Changed these to check for emptiness.

Reply:
But we initialize the array:

```ts
let chromeValues: T[] = [];
```

and in JS even an empty array will always be `true`:

```js
Boolean([]) // true
Boolean([1,2,3]) //true
```

so there is no point in this check) If you mean checking for an empty array then it should be `chromeValues.length`